### PR TITLE
7030853: JDK 7 Serializable Swing classes not compatible with JDK 6

### PR DESCRIPTION
**Pull request by prsadhuk**
The serialized form of the following classes in JDK 7 builds are not compatible with prior versions of Java:
    javax.swing.text.html.parser.ParserDelegator
    javax.swing.LayoutComparator
They need to be declared as not compatible between releases in the API documentation, as is the case for some other serializable Swing classes.

Since It is widely known that serialized Swing classes are not comptible between major releases, we just didn't add the warning at that couple classes.
Added the warning..

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires a CSR request to be approved

### Issues
 * [JDK-7030853](https://bugs.openjdk.org/browse/JDK-7030853): JDK 7 Serializable Swing classes not compatible with JDK 6
 * [JDK-8298431](https://bugs.openjdk.org/browse/JDK-8298431): JDK 7 Serializable Swing classes not compatible with JDK 6 (**CSR**)

### Reviewers
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**) ⚠️ Review applies to [6ac67d2f](https://git.openjdk.org/jdk/pull/11565/files/6ac67d2f751e86e1a77eecfbb373c3cf2a5ded36)
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11565/head:pull/11565` \
`$ git checkout pull/11565`

Update a local copy of the PR: \
`$ git checkout pull/11565` \
`$ git pull https://git.openjdk.org/jdk pull/11565/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11565`

View PR using the GUI difftool: \
`$ git pr show -t 11565`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11565.diff">https://git.openjdk.org/jdk/pull/11565.diff</a>

</details>

### CODE REVIEW

**Comment by mrserb** [comment]
I think we will need a CSR for this change.  What people think about adding this note to the documentation of the swing package?

**Comment by prrace** [comment]
Is this something that should be handled like the threading policy where the javax.swing package doc spells it out for the entire API ?

**Comment by prsadhuk** [comment]
OK. Added the same in javax.swing package..
Retained individual class too as the warnings are present in other classes too

**Comment by prsadhuk** [comment]
CSR https://bugs.openjdk.org/browse/JDK-8298431
Please review

File: src/java.desktop/share/classes/javax/swing/package-info.java [hunk 1]
```diff
@@ -128,7 +128,17 @@
  * href="https://docs.oracle.com/javase/tutorial/uiswing/concurrency/index.html">
  * Concurrency in Swing</a>.
  *
+ * <h2><a id="serialization">Swing's Serialization policy</a></h2>
+ * <strong>Warning:</strong>
+ * Serialized objects of any swing class will not be compatible with
```
Comment (prrace) [inline src/java.desktop/share/classes/javax/swing/package-info.java:133]: capitalise swing

File: src/java.desktop/share/classes/javax/swing/text/html/parser/ParserDelegator.java [hunk 2]
```diff
@@ -42,7 +42,13 @@
...
  * each time its parse method is invoked. Stores a
  * reference to the dtd.
- *
+ * <strong>Warning:</strong>
+ *  Serialized objects of this class will not be compatible with
+ *  future Swing releases. The current serialization support is
+ *  appropriate for short term storage or RMI between applications running
+ *  the same version of Swing.  As of 1.4, support for long term storage
+ *  of all JavaBeans
+ *  has been added to the <code>java.beans</code> package
```
Comment (mrserb) [inline src/java.desktop/share/classes/javax/swing/text/html/parser/ParserDelegator.java:51]: Looks like it has an unnecessary space after `*`

File: src/java.desktop/share/classes/javax/swing/text/html/parser/ParserDelegator.java [hunk 3]
```diff
@@ -42,6 +42,12 @@
  * Responsible for starting up a new DocumentParser
  * each time its parse method is invoked. Stores a
  * reference to the dtd.
+ * <strong>Warning:</strong>
```
Comment (aivanov-jdk) [inline src/java.desktop/share/classes/javax/swing/text/html/parser/ParserDelegator.java:45]: Missing `<p>`.

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]

**Review by aivanov-jdk — APPROVED** [review]

**Comment by aivanov-jdk** [comment]
Does this change require updating the CSR?

**Pull request closed without merging**